### PR TITLE
Feature/app header

### PR DESCRIPTION
**Pull request by YuvalLeviMoveoHLS**

### CODE REVIEW

File: src/components/AppHeader/appHeader.tsx [hunk 1]
```diff
@@ -1,20 +1,44 @@
...
-      {/* <div style={{ "background-color": "black" }}>test</div> */}
-      {/* <FilterDropDown
-        selectOptions={selectOptionsAppHeader}
-        placeholder="Top Headlines"
-        isAppHeader={true}
-      ></FilterDropDown> */}
+
+      <StyledInputWrapper>
+        <SearchInput showFilterDropDown={true} onClick={toggleDropdown} />
+        {isDropdownVisible && <RecentSearchDropdown ref={dropdownRef} />}
```
Comment (ZivCohenMoveo) [inline src/components/AppHeader/appHeader.tsx]: put it together in component of `DropdownWithSearch` and change the `RecentSearchDropdown` name to `DropdownBody` or something like this

Comment (ZivCohenMoveo) [reply src/components/AppHeader/appHeader.tsx]: And move all the logic in lines 13-33 inside `DropdownWithSearch` component

File: src/components/AppHeader/appHeader.stories.ts
Comment (ZivCohenMoveo) [inline src/components/AppHeader/appHeader.stories.ts:1]: Components name should be in title case
Fix it in all files

File: src/components/AppHeader/appHeader.tsx [hunk 2]
```diff
@@ -1,20 +1,44 @@
...
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const toggleDropdown = () => {
+    setDropdownVisibility(!isDropdownVisible);
+  };
+  const handleClickOutside = (event: MouseEvent) => {
+    if (
+      dropdownRef.current &&
+      !dropdownRef.current.contains(event.target as Node)
+    ) {
+      console.log(dropdownRef.current.contains(event.target as Node));
```
Comment (ZivCohenMoveo) [inline src/components/AppHeader/appHeader.tsx]: remove

File: src/components/Input/searchInput.tsx [hunk 3]
```diff
@@ -5,23 +5,36 @@ import {
...
+        placeholder={placeholder}
+        onClick={onClick}
+        onFocus={(e) => (e.target.placeholder = "")}
+      />
+      {showFilterDropDown && (
+        <FilterDropDown
+          selectOptions={selectOptionsAppHeader}
+          placeholder="Top Headlines"
+          isAppHeader={true}
+        ></FilterDropDown>
```
Comment (ZivCohenMoveo) [inline src/components/Input/searchInput.tsx]: Shouldn't be inside SearchInput
SearchInput should be only the input itself
The input and the dropdown together should be in different component like `DropdownWithSearch`

File: src/components/RecentSearchDropdown/RecentSearchDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  searches?: string[];
+}
+
+const RecentSearchDropdown: React.ForwardRefRenderFunction<
+  HTMLDivElement,
+  RecentSearchDropdownProps
+> = (props, ref) => {
+  const [recentSearches, setRecentSearches] = useState<string[]>(
+    props.searches || recentSearchesMock
+  );
```
Comment (ZivCohenMoveo) [inline src/components/RecentSearchDropdown/RecentSearchDropdown.tsx]: You don't need useState.

```suggestion
  const recentSearches = props.searches ?? recentSearchesMock
```

File: src/components/RecentSearchDropdown/RecentSearchDropdown.stories.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  component: RecentSearchDropdown,
+  //tags: ["autodocs"],
+} satisfies Meta<typeof RecentSearchDropdown>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const RecentSearchStory: Story = {
+  args: {
+    searches: ["crypto", "soccer", "basketball", "NBA"],
```
Comment (ZivCohenMoveo) [inline src/components/RecentSearchDropdown/RecentSearchDropdown.stories.ts]: use `recentSearchesMock`

File: src/utils/constants/maxValues.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export const MAX_AMOUNT_RECENT_SEARCHES = 5;
```
Comment (ZivCohenMoveo) [inline src/utils/constants/maxValues.ts]: in use?

File: src/styles/globalStyles.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import styled from "styled-components";
+
+export const AppContainer = styled.div`
```
Comment (ZivCohenMoveo) [inline src/styles/globalStyles.ts]: move it to `App.styles.ts`

**Review by ZivCohenMoveo — CHANGES_REQUESTED** [review]

**Review by ronc1812 — APPROVED** [review]

**Pull request merged**